src/goto-symex/symex_goto.cpp

```diff
@@ -319,14 +326,21 @@ void goto_symext::merge_goto(
       "atomic sections differ across branches",
       state.source.pc->source_location);
 
-  // do SSA phi functions
-  phi_function(goto_state, state);
+  if(!goto_state.guard.is_false())
+  {
+    if(state.guard.is_false())
+    {
+      state.move_from(std::move(goto_state));
```

Comment:
I think this is now replacable by the default move-assign op? (Again comment that this is our superclass being moved)

Reply:
Kinda, see the last commit for one way to do it.

Reply:
That looks good to me -- it'd be cleaner if it was a member not a superclass, but otherwise as hoped for.